fs-repo-14-to-15/migration/migration.go

```diff
@@ -172,65 +172,79 @@ func convert(in io.Reader, out io.Writer) error {
 	}
 
 	// Remove /quic only addresses from .Addresses.Swarm
-	if a, ok := confMap["Addresses"]; ok {
-		addresses, ok := a.(map[string]any)
-		if !ok {
-			return fmt.Errorf("invalid type for .Addresses got %T expected json map", a)
-		}
-
-		for _, addressToRemove := range [...]string{"Swarm", "Announce", "AppendAnnounce", "NoAnnounce"} {
-			s, ok := addresses[addressToRemove]
+	if err := func() error {
+		if a, ok := confMap["Addresses"]; ok {
+			addresses, ok := a.(map[string]any)
 			if !ok {
-				continue
+				fmt.Printf("invalid type for .Addresses got %T expected json map; skipping .Addresses\n", a)
+				return nil
 			}
 
-			swarm, ok := s.([]interface{})
-			if !ok {
-				return fmt.Errorf("invalid type for .Addresses.%s got %T expected json array", addressToRemove, s)
-			}
+			for _, addressToRemove := range [...]string{"Swarm", "Announce", "AppendAnnounce", "NoAnnounce"} {
+				s, ok := addresses[addressToRemove]
+				if !ok {
+					continue
+				}
+
+				swarm, ok := s.([]interface{})
+				if !ok {
+					fmt.Printf("invalid type for .Addresses.%s got %T expected json array; skipping .Addresses.%s\n", addressToRemove, s, addressToRemove)
+					continue
+				}
 
-			var newSwarm []interface{}
-			for _, v := range swarm {
-				if addr, ok := v.(string); ok {
-					if !strings.Contains(addr, "/quic") || strings.Contains(addr, "/quic-v1") {
-						newSwarm = append(newSwarm, addr)
+				var newSwarm []interface{}
+				for _, v := range swarm {
+					if addr, ok := v.(string); ok {
+						if !strings.Contains(addr, "/quic") || strings.Contains(addr, "/quic-v1") {
+							newSwarm = append(newSwarm, addr)
+						}
+					} else {
+						newSwarm = append(newSwarm, v)
```

Comment:
I know this was here before, but what is this if not a string? Are we just keeping it so as not to interfere?

Reply:
yes